lisa/tools/ethtool.py

```diff
@@ -258,46 +258,62 @@ def _get_device_list(self) -> None:
             if not netdir:
                 continue
             cmd_result = self.node.execute(f"ls {netdir}")
-            if cmd_result.exit_code != 0:
-                raise LisaException(
-                    "Could not find the network device under path {netdir}"
-                    f" exit_code: {cmd_result.exit_code}"
-                    f" stderr: {cmd_result.stderr}"
-                )
-            self._device_list.append(cmd_result.stdout)
+            cmd_result.assert_exit_code(
+                message=f"Could not find the network device under path {netdir}"
+            )
 
-        if not self._device_list:
+            self._device_set.add(cmd_result.stdout)
+
+        if not self._device_set:
             raise LisaException("Did not find any synthetic network interface.")
 
+        return self._device_set
+
     def get_device_channels_info(
-        self,
-        interface: str,
+        self, interface: str, force: bool = False
     ) -> DeviceChannel:
-        if interface in self._device_channel_map.keys():
+        if (not force) and (interface in self._device_channel_map.keys()):
             return self._device_channel_map[interface]
 
-        result = self.run(f"-l {interface}")
-        if result.exit_code != 0:
-            raise LisaException(
-                f"ethtool -l {interface} command got non-zero"
-                f" exit_code: {result.exit_code}"
-                f" stderr: {result.stderr}"
-            )
+        result = self.run(f"-l {interface}", force_run=force)
+        if (result.exit_code != 0) and ("Operation not supported" in result.stderr):
+            raise SkippedException
```

Comment:
1. The `SkippedException` should be in test case, not in tools, because the Skipped is a result for test case. The tool may be used in different test cases.
2. The reason must be specified, why it's skipped.

Reply:
You can raise a LISAException or a new type of Exception, and capture it in test case, and covert it to SkippedException. The two exceptions have different purpose and message, the inner exception is to specify the command failed, the skipped exception explain something doesn't support, so the test case is skipped.